cpp/mrc/include/mrc/coroutines/task_container.hpp

```diff
@@ -166,6 +172,10 @@ class TaskContainer
     std::shared_ptr<Scheduler> m_scheduler_lifetime{nullptr};
     /// This is used internally since io_scheduler cannot pass itself in as a shared_ptr.
     Scheduler* m_scheduler{nullptr};
+    /// tasks to be processed in order of start
+    std::queue<decltype(m_tasks.end())> m_next_tasks;
+    /// maximum number of tasks to be run simultaneously
+    int32_t m_max_simultaneous_tasks;
```

Comment:
Why not have this as a `size_t` like the constructor signature and use 0 for unlimited number of tasks? A value of 0 would be invalid so we either need to check for that situation or just use 0 for the unlimited use case.

Reply:
Ah, I just forgot to change this to size_t. I think allowing any zero-or-negative number for unlimited tasks makes sense, otherwise we have to throw on negatives. size_t supports negatives.

Reply:
I forgot to change this, and I always seem to forget that size_t is unsigned.